dftimewolf/lib/exporters/timesketch.py

```diff
@@ -152,67 +156,49 @@ def _GetSketchIDFromAttributes(self) -> int:
           return sketch_id
     return 0
 
-  def Process(self) -> None:
-    """Executes a Timesketch export."""
-    if not self.timesketch_api:
-      message = 'Could not connect to Timesketch server'
-      self.ModuleError(message, critical=True)
+  def Process(self, container: containers.File) -> None:
+    """Executes a Timesketch export.
 
-    sketch = self.state.GetFromCache('timesketch_sketch')
-    if not sketch and self.sketch_id:
-      self.logger.info('Using exiting sketch: {0:d}'.format(self.sketch_id))
-      sketch = self.timesketch_api.get_sketch(self.sketch_id)
-
-    # Create the sketch if no sketch was stored in the cache.
-    if not sketch:
-      sketch = self._CreateSketch(incident_id=self.incident_id)
-      self.sketch_id = sketch.id
-      self.logger.info('New sketch created: {0:d}'.format(self.sketch_id))
+    Args:
+      container (containers.File): A container holding a File to import."""
 
     recipe_name = self.state.recipe.get('name', 'no_recipe')
-    input_names = []
 
-    for file_container in self.state.GetContainers(containers.File):
-      description = file_container.name
-      if not description:
-        continue
+    description = container.name
+    if description:
       name = description.rpartition('.')[0]
+      name = name if name else description
       name = name.replace(' ', '_').replace('-', '_')
-      input_names.append(name)
-
-    if input_names:
       timeline_name = '{0:s}_{1:s}'.format(
-          recipe_name, '_'.join(input_names))
+          recipe_name, name)
     else:
       timeline_name = recipe_name
 
     with importer.ImportStreamer() as streamer:
-      streamer.set_sketch(sketch)
+      streamer.set_sketch(self.sketch)
       streamer.set_timeline_name(timeline_name)
 
-      for file_container in self.state.GetContainers(containers.File):
-        path = file_container.path
-        streamer.add_file(path)
-        if streamer.response and file_container.description:
-          streamer.timeline.description = file_container.description
+      path = container.path
+      streamer.add_file(path)
+      if streamer.response and container.description:
+        streamer.timeline.description = container.description
 
-    api_root = sketch.api.api_root
-    host_url = api_root.partition('api/v1')[0]
-    sketch_url = '{0:s}sketches/{1:d}/'.format(host_url, sketch.id)
+    api_root = self.sketch.api.api_root
+    self.host_url = api_root.partition('api/v1')[0]
+    sketch_url = '{0:s}sketches/{1:d}/'.format(self.host_url, self.sketch.id)
     message = 'Your Timesketch URL is: {0:s}'.format(sketch_url)
-    self.logger.success(message)
-    container = containers.Report(
+    report_container = containers.Report(
         module_name='TimesketchExporter',
         text=message,
         text_format='markdown')
-    self.state.StoreContainer(container)
+    self.state.StoreContainer(report_container)
```

Comment:
Given that we'll be pushing things to the same sketch, wouldn't it make sense to store just a single report?

Reply:
I don't think I agree, the source of this could be multiple GRR collections, multiple turbinia runs, etc.... I think one report per input timeline makes sense.

Reply:
Yeah, makes sense. Pushed to `PostProcess`.